pkg/reconciler/tenancy/workspacemounts/workspacemounts_controller.go

```diff
@@ -0,0 +1,302 @@
+		c.queue.Add(key)
+		return true
+	}
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *Controller) process(ctx context.Context, key string) (bool, error) {
+	if strings.HasPrefix(key, workspaceKeyPrefix) {
+		return c.processWorkspace(ctx, key)
+	}
+	if strings.HasPrefix(key, gvrKeyPrefix) {
+		return c.processGVKMount(ctx, key)
+	}
+	return false, fmt.Errorf("unknown key prefix: %s", key)
+}
+
+func (c *Controller) processWorkspace(ctx context.Context, key string) (bool, error) {
+	parent, name, err := parseWorkspaceKey(key)
+	if err != nil {
+		runtime.HandleError(err)
+		return false, nil
+	}
+	workspace, err := c.workspaceLister.Cluster(parent).Get(name)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return false, nil // object deleted before we handled it
+		}
+		return false, err
+	}
+
+	old := workspace
+	workspace = workspace.DeepCopy()
+
+	logger := logging.WithObject(klog.FromContext(ctx), workspace)
+	ctx = klog.NewContext(ctx, logger)
+
+	var errs []error
+	requeue, err := c.reconcile(ctx, workspace)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	// If the object being reconciled changed as a result, update it.
+	oldResource := &workspaceResource{ObjectMeta: old.ObjectMeta, Spec: &old.Spec, Status: &old.Status}
+	newResource := &workspaceResource{ObjectMeta: workspace.ObjectMeta, Spec: &workspace.Spec, Status: &workspace.Status}
+	if err := c.commit(ctx, oldResource, newResource); err != nil {
+		errs = append(errs, err)
+	}
+
+	return requeue, utilerrors.NewAggregate(errs)
+}
+
+func (c *Controller) processGVKMount(ctx context.Context, key string) (bool, error) {
+	logger := klog.FromContext(ctx)
+
+	gvr, key, err := parseGVKKey(key)
+	if err != nil {
+		runtime.HandleError(err)
+		return false, nil
+	}
+
+	logger = logger.WithValues("gvr", gvr.String(), "name", key)
+
+	inf, err := c.discoveringDynamicSharedInformerFactory.ForResource(gvr)
+	if err != nil {
+		return false, fmt.Errorf("error getting dynamic informer for GVR %q: %w", gvr, err)
+	}
+
+	maybeMountObj, exists, err := inf.Informer().GetIndexer().GetByKey(key)
+	if err != nil {
+		logger.Error(err, "unable to get from indexer")
+		return false, nil // retrying won't help
+	}
+	if !exists {
+		logger.V(4).Info("resource not found")
+		return false, nil
+	}
+
+	u, ok := maybeMountObj.(*unstructured.Unstructured)
+	if !ok {
+		logger.Error(nil, "got unexpected type", "type", fmt.Sprintf("%T", maybeMountObj))
+		return false, nil // retrying won't help
+	}
+
+	u = u.DeepCopy()
+	if u.GetAnnotations() == nil {
+		return false, nil
+	}
+
+	// We only care about mount objects. All owner objects must have IsMountAnnotationKey set
+	// to simplify the logic here. And owner annotation to owner workspaces.
+	val, ok := u.GetAnnotations()[tenancyv1alpha1.ExperimentalIsMountAnnotationKey]
+	if !ok || val != "true" {
+		return false, nil
+	}
+
+	workspaceOwnerRaw, ok := u.GetAnnotations()[tenancyv1alpha1.ExperimentalWorkspaceOwnerAnnotationKey]
```

Comment:
the use of this annotation irritates me. We usually use it to store the userInfo of the user that created a workspace.

Reply:
should we add a dedicated one? 
`experimental.tenancy.kcp.io/owning-workspace`

Reply:
agreed. lets move to index post kubecon? Else I need to redo our demo env :D